packages/base/src/dialogs/symbology/hooks/useGetSingleBandInfo.ts

```diff
@@ -0,0 +1,150 @@
+import { IJGISLayer, IJupyterGISModel } from '@jupytergis/schema';
+import { fromUrl, fromBlob } from 'geotiff';
+import { useEffect, useState } from 'react';
+
+import { loadFile } from '@/src/tools';
+
+export interface IBandRow {
+  band: number;
+  colorInterpretation?: string;
+  stats: {
+    minimum: number;
+    maximum: number;
+  };
+}
+
+const useGetSingleBandInfo = (
+  model: IJupyterGISModel,
+  layer: IJGISLayer,
+  layerId: string,
+  selectedBand: number,
+) => {
+  const [bandRows, setBandRows] = useState<IBandRow[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchBandInfo = async () => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      const bandsArr: IBandRow[] = [];
+      const source = model.getSource(layer?.parameters?.source);
+      const sourceInfo = source?.parameters?.urls[0];
+
+      if (!sourceInfo?.url) {
+        setError('No source URL found.');
+        setLoading(false);
+        return;
+      }
+
+      let tiff;
+      if (
+        sourceInfo.url.startsWith('http') ||
+        sourceInfo.url.startsWith('https')
+      ) {
+        // Handle remote GeoTIFF file
+        tiff = await fromUrl(sourceInfo.url);
+      } else {
+        // Handle local GeoTIFF file
+        const preloadedFile = await loadFile({
+          filepath: sourceInfo.url,
+          type: 'GeoTiffSource',
+          model,
+        });
+
+        if (!preloadedFile.file) {
+          setError('Failed to load local file.');
+          setLoading(false);
+          return;
+        }
+
+        tiff = await fromBlob(preloadedFile.file);
+      }
+
+      const image = await tiff.getImage();
+      const numberOfBands = image.getSamplesPerPixel();
+
+      let min = layer.parameters?.symbologyState?.min;
+      let max = layer.parameters?.symbologyState?.max;
+
+      if (min === undefined || max === undefined) {
+        // 1. Try metadata first
+        let dataMin = image.fileDirectory.STATISTICS_MINIMUM;
+        let dataMax = image.fileDirectory.STATISTICS_MAXIMUM;
+
+        if (dataMin === undefined || dataMax === undefined) {
+          // 2. Try smallest overview if available
```

Comment:
```suggestion
          // 2. Try smallest overview if available
          // A consequence of this method is that we will almost never find the **true** minimum and maximum for the whole raster, as overviews are aggregate data, averaging groups of grid cells to calculate lower-resolution grid cells. However, calculating the minimum and maximum from the entire dataset may be prohibitively expensive on CPU and memory!
```

There's probably a better way to do this, but we don't need to solve that now. We just need to document the tradeoff we're making :)